engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -799,6 +807,10 @@ public void monitorHeartbeat()
     processEvents();
 
     committedWindowId = updateCheckpoints(false);
+    if (lastCommittedWindowId != committedWindowId) {
```

Comment:
yes, this is added so that multiple call don't happen for same committed windowId.